torchrl/modules/models/models.py

```diff
@@ -1135,6 +1136,116 @@ def forward(
+        transformer_config: Dict,
+        device: Optional[DEVICE_TYPING] = None,
+    ):
+        super().__init__()
+        self.transformer = DecisionTransformer(
+            state_dim=state_dim,
+            action_dim=action_dim,
+            config=transformer_config,
+        )
+        self.action_layer = nn.Linear(
+            transformer_config.n_embd, action_dim * 2, device=device
+        )
+
+        self.log_std_min, self.log_std_max = -5.0, 2.0
+
+        def weight_init(m):
+            """Custom weight init for Conv2D and Linear layers."""
+            if isinstance(m, torch.nn.Linear):
+                nn.init.orthogonal_(m.weight.data)
+                if hasattr(m.bias, "data"):
+                    m.bias.data.fill_(0.0)
+
+        self.apply(weight_init)
+
+    def forward(
+        self,
+        observation: torch.Tensor,
+        action: torch.Tensor,
+        return_to_go: torch.Tensor,
+    ) -> torch.Tensor:
+        if observation.ndim == 2:
+            observation = observation.unsqueeze(0)
+            action = action.unsqueeze(0)
+            return_to_go = return_to_go.unsqueeze(0)
+        hidden_state = self.transformer(observation, action, return_to_go)
+        out = self.action_layer(hidden_state)
+        mu, log_std = torch.chunk(out, 2, -1)
+        log_std = torch.tanh(log_std)
+        # log_std is the output of tanh so it will be between [-1, 1]
+        # map it to be between [log_std_min, log_std_max]
+        log_std = self.log_std_min + 0.5 * (self.log_std_max - self.log_std_min) * (
+            log_std + 1.0
+        )
+        std = log_std.exp()
+
+        return (mu, std)
+
+
+class DTActor(nn.Module):
+    """Decision Transformer Actor class.
+
+    Presented in "Decision Transformer",
+    https://arxiv.org/abs/2202.05607.pdf
+
+
+    """
+
+    def __init__(
+        self,
+        state_dim: int,
+        action_dim: int,
+        transformer_config: Dict,
+        device: Optional[DEVICE_TYPING] = None,
+    ):
+        super().__init__()
+        self.transformer = DecisionTransformer(
+            state_dim=state_dim,
+            action_dim=action_dim,
+            config=transformer_config,
+        )
+        self.action_layer = nn.Linear(
+            transformer_config.n_embd, action_dim, device=device
+        )
+
+        def weight_init(m):
+            """Custom weight init for Conv2D and Linear layers."""
+            if isinstance(m, torch.nn.Linear):
+                nn.init.orthogonal_(m.weight.data)
+                if hasattr(m.bias, "data"):
+                    m.bias.data.fill_(0.0)
+
+        self.apply(weight_init)
+
+    def forward(
+        self,
+        observation: torch.Tensor,
+        action: torch.Tensor,
+        return_to_go: torch.Tensor,
+    ) -> torch.Tensor:
+        if observation.ndim == 2:
+            observation = observation.unsqueeze(0)
+            action = action.unsqueeze(0)
+            return_to_go = return_to_go.unsqueeze(0)
```

Comment:
not sure it makes sense so feel free to ignore
shouldn't we squeeze the output in those cases?